extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/conversational.py

```diff
@@ -0,0 +1,285 @@
+    Api Ref: https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/inference/_client.py#L338
+    Doc Ref: https://huggingface.co/docs/huggingface_hub/package_reference/inference_client#huggingface_hub.InferenceClient.conversational
+    """
+
+    supported_keys = {
+        "model",
+    }
+
+    completion_data: dict[str, Any] = {}
+
+    for key in model_settings:
+        if key.lower() in supported_keys:
+            completion_data[key.lower()] = model_settings[key]
+
+    if "model" in completion_data:
+        # HF is in the process of revamping their libraries. Default behaviour of Converstaional is broken
+        # This temporary fix is reccomended by HF https://github.com/huggingface/huggingface_hub/issues/2023#issuecomment-1935760040
+        completion_data["model"] = (
+            f"https://api-inference.huggingface.co/pipeline/conversational/{completion_data['model']}"
+        )
+    else:
+        # manually set default model because, otherwise this will fail. see comment above.
+        # see this thread for more info: https://github.com/huggingface/huggingface_hub/issues/2023
+        completion_data["model"] = "https://api-inference.huggingface.co/models/facebook/blenderbot-400M-distill"
+
+    return completion_data
+
+class HuggingFaceConversationalRemoteInference(ParameterizedModelParser):
+    """
+    A model parser for HuggingFace conversational models.
+    """
+
+    def __init__(self, model_id: str = None, use_api_token=False):
+        """
+        Args:
+            model_id (str): The model ID of the model to use.
+            no_token (bool): Whether or not to require an API token. Set to False if you don't have an api key.
+
+        Returns:
+            HuggingFaceConversationalRemoteInference: The HuggingFaceConversationalRemoteInference object.
+
+        Usage:
+
+        1. Create a new model parser object with the model ID of the model to use.
+                parser = HuggingFaceConversationalRemoteInference("facebook/blenderbot-400M-distill", use_api_token=False)
+        2. Add the model parser to the registry.
+                config.register_model_parser(parser)
+
+        If use_api_token is set to True, then the model parser will require an API token to be set in the environment variable HUGGING_FACE_API_TOKEN.
+
+
+        """
+        super().__init__()
+
+        token = None
+
+        if use_api_token:
+            # You are allowed to use Hugging Face for a bit before you get
+            # rate limited, in which case you will receive a clear error
+            token = get_api_key_from_environment(
+                "HUGGING_FACE_API_TOKEN", required=False
+            ).unwrap()
+
+        self.client = InferenceClient(model_id, token=token)
+
+    def id(self) -> str:
+        """
+        Returns an identifier for the Model Parser
+        """
+        return "HuggingFaceConversationalRemoteInference"
```

Comment:
Checking if this is the name that will show up in the UI or if it can be adjusted to be more user friendly (I.e. "Chat")

Reply:
This won't show up as "HuggingFaceConversationalRemoteInference" inside Gradio Notebook. The Gradio Notebook has a customized model parser registry.